Enhance bindings

Nicogene opened the pull request:
This PR cleans up yarp.i and integrates and extends ruby and python tests with ctest

Please review code.

traversaro commented on bindings/python/tests/CMakeLists.txt:
```diff
@@ -4,4 +4,22 @@
 
-# TODO
+macro(add_python_unit_test pythonscript)
+  find_program(PYTHON_EXECUTABLE
```
`PYTHON_EXECUTABLE` is already defined in parent scope, also respecting the `YARP_USE_PYTHON_VERSION`, so I would avoid the `find_program` here (note tha this is different from Lua because the FindLua.cmake module in CMake does not search for the lua interpreter, so we need to search it autonomously).

traversaro commented on bindings/ruby/tests/CMakeLists.txt:
```diff
@@ -4,4 +4,19 @@
 
-# TODO
+macro(add_ruby_unit_test rubyscript)
+  find_program(RUBY_EXECUTABLE
```
`RUBY_EXECUTABLE` should be already defined by the call to `find_package(Ruby)` (see similar comment in the python bindings).

traversaro commented on bindings/lua/tests/CMakeLists.txt:
```diff
@@ -16,7 +16,10 @@ macro(add_lua_unit_test luascript)
-add_lua_unit_test(test_vocab.lua)
-add_lua_unit_test(test_port.lua)
+
+file(GLOB yarp_lua_test_files "${CMAKE_CURRENT_SOURCE_DIR}/*.lua")
```
I do not have a strong opinion on this, but typically on YARP we avoid CMake globbing (see https://stackoverflow.com/questions/32411963/why-is-cmake-file-glob-evil). I guess @drdanz has the last word here.

Nicogene replied:
I just copied what there is in other tests(harness)
https://github.com/robotology/yarp/blob/d602780457686d2536cd101fb9f11a0b1621ccf3/tests/CMakeLists.txt#L97-L98

The alternative of globbing is to add each test one by one ??

traversaro replied:
> The alternative of globbing is to add each test one by one ??

Exactly. That's the reason why I do not have a strong opinion on this. Listing all the files is convenient for the reasons outline in https://stackoverflow.com/questions/32411963/why-is-cmake-file-glob-evil , but it is sort of duplicating the presence of the files with a given directory in the filesystem.

jgvictores commented on bindings/python/tests/test_buffered_port.py:
```diff
@@ -0,0 +1,48 @@
+
+
+yarp.Network.init()
+
```
A `yarp.Network.setLocalMode(True)` line here would avoid warnings.

jgvictores commented on bindings/ruby/tests/test_buffered_port.rb:
```diff
@@ -0,0 +1,41 @@
+  end
+end
+
+Yarp::Network.init()
```
A `Yarp::Network.setLocalMode(true)` line here (not sure about Ruby syntax) would avoid warnings.

Nicogene commented:
This PR has been moved to master because we need this integration with ctest to debug the bindings

Nicogene commented:
The PR is ready to be merged.
Additionally I added for lua python and ruby a test on yarp images(setExternal) and I added the install target in ruby bindings accordingly how ruby is installed in the system

Nicogene commented:
As soon as this PR passes I will merge it, ok ? @drdanz @traversaro

traversaro commented:
Yes!
